### PR TITLE
Issue #495: Fix dry run issue in curl connectors

### LINKED ISSUE #495: connector_curl should still read from the site in a dry run when there is no side effect.

### PR DESCRIPTION
**Pull request by jaypha**
Resolves #495

### CODE REVIEW

File: classes/local/step/connector_curl.php [hunk 1]
```diff
@@ -250,7 +250,7 @@ public function execute(): bool {
         }
 
         // Download response to file provided destination is set.
-        if (!empty($config->destination) && !$isdryrun) {
+        if (!empty($config->destination) && $doingcurl) {
```
Comment (brendanheywood) [inline classes/local/step/connector_curl.php:253]: can we make this logic simpler and just exit earlier if it isn't curling?

Comment (jaypha) [reply classes/local/step/connector_curl.php]: done

File: classes/local/step/connector_curl.php [hunk 2]
```diff
@@ -30,6 +30,12 @@
  */
 class connector_curl extends connector_step {
 
+    /** Error number for curl timeout. */
+    public const CURL_TIMEOUT_ERROR = 28;
```
Comment (brendanheywood) [inline classes/local/step/connector_curl.php:34]: can this use the php curl constants? is this the same as CURLE_OPERATION_TIMEOUTED

Comment (jaypha) [reply classes/local/step/connector_curl.php]: done

**Pull request merged**